app/preview.py

```diff
@@ -32,25 +32,42 @@ def hide_notify_tag(image):
 
 @statsd(namespace="template_preview")
 def png_from_pdf(data, page_number, hide_notify=False):
-    output = BytesIO()
     with Image(blob=data, resolution=150) as pdf:
         pdf_width, pdf_height = pdf.width, pdf.height
         try:
             page = pdf.sequence[page_number - 1]
         except IndexError:
             abort(400, 'Letter does not have a page {}'.format(page_number))
         pdf_colorspace = pdf.colorspace
+    return _generate_png_page(page, pdf_width, pdf_height, pdf_colorspace, hide_notify)
+
+
+@statsd(namespace="template_preview")
+def pngs_from_pdf(data):
+    pages = []
+    with Image(blob=data, resolution=150) as pdf:
+        pdf_width, pdf_height = pdf.width, pdf.height
+        pdf_colorspace = pdf.colorspace
+        for page in pdf.sequence:
+            output = _generate_png_page(page, pdf_width, pdf_height, pdf_colorspace)
+            pages.append(output)
+
+    return pages
+
+
+@statsd(namespace="template_preview")
+def _generate_png_page(pdf_page, pdf_width, pdf_height, pdf_colorspace, hide_notify=False):
+    output = BytesIO()
     with Image(width=pdf_width, height=pdf_height) as image:
 
         if pdf_colorspace == 'cmyk':
             image.transform_colorspace('cmyk')
 
-        image.composite(page, top=0, left=0)
+        image.composite(pdf_page, top=0, left=0)
         if hide_notify:
             hide_notify_tag(image)
         converted = image.convert('png')
```

Comment:
This isn't related to our changes, but I think this should be `with image.convert('png') as converted` since that's what most `wand` examples are using. As far as I understand it might help avoid memory leaks if something fails during the conversion.

Reply:
Ahh, cool, well-spotted 👍 . I will change it ☺️